ghcide/test/exe/Main.hs

```diff
@@ -6075,11 +6075,14 @@ ifaceErrorTest = testCase "iface-error-test-1" $ runWithExtraFiles "recomp" $ \d
     expectDiagnostics
       [("P.hs", [(DsWarning,(4,0), "Top-level binding")])] -- So what we know P has been loaded
 
+    waitForProgressDone
```

Comment:
Worth some comments explaining that these calls allows the changes to settle?

Reply:
The 100s of existing unit tests implicitly document this approach. Even if I added a few comments here, they would not be discoverable enough to be helpful